### PR TITLE
[Rebass] Update rebass base props to facilitate use of string or css function

### PR DESCRIPTION
**Pull request by jbarzegar**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<https://www.styled-components.com/docs/api#css-prop>>
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

Hello! I noticed rebass complained whenever a string is used inside its css prop, however as the [styled-components docs show](https://www.styled-components.com/docs/api#css-prop) you can use a string/template literal.

### CODE REVIEW

**Review by angusfretwell — APPROVED** [review]
Good catch!

**Review by rhysd — APPROVED** [review]

File: types/rebass/index.d.ts [hunk 1]
```diff
@@ -18,7 +18,10 @@ type Omit<T, K> = Pick<T, Exclude<keyof T, K>>;
 
 export interface BaseProps extends React.Props<any> {
     as?: React.ReactType;
-    css?: StyledComponents.CSSObject;
+    css?:
+        | StyledComponents.CSSObject
+        | StyledComponents.BaseThemedCssFunction<any>
```
Comment (rhysd) [inline types/rebass/index.d.ts]: Though it is not mandatory, adding test case for `StyledComponents.BaseThemedCssFunction` would make this patch even better.

Comment (jbarzegar) [reply types/rebass/index.d.ts]: I'll add a test case right now.

Comment (jbarzegar) [reply types/rebass/index.d.ts]: Ah whoops I realized that `StyledComponents.BaseThemedCssFunction` isn't returned to the `css` prop but `StyledComponents.FlattenSimpleInterpolation` is.  Will fix

Comment (jbarzegar) [reply types/rebass/index.d.ts]: Fixed

**Review by rhysd — APPROVED** [review]
Thank you. Looks good

**Pull request merged**